openapi/platform/platform.yaml

```diff
@@ -1245,6 +1246,38 @@ components:
                     description: The billing frequency
                 required:
                   - sku
+            statusCode:
+              type: string
+              default: draft
+              enum:
+                - submitted
+                - approved
+                - declined
+                - draft
+                - processing
+                - error
+                - archived
+                - submittedForCustomerApproval
+                - cancellationRequested
+                - canceled
+                - scheduled
+                - processingPayment
+                - resubmitted
+              description: |-
+                The order status of the order will be one of the following:
+                - `submitted`: The order is ready for admin approval
+                - `approved`: The order has been approved by an admin
+                - `declined`: The order has been declined by an admin or a customer
```

Comment:
I don't suppose it is feasible to know if it was the admin or customer that declined the order? That would impact where the user wants to send it if they resubmit the order. 

If we could reasonably map the values to our internal data model it would be idea to split the status into an overall order status, admin approval status and a customer approval status. With the ability for resubmission and the approval coming in different orders I find the current list a little confusing. 
I think that would make sense along side displaying the user id and approval time for the admin and customer who last approved.

Reply:
Is there any slice that is feasible or statuses we can omit/rename? If we find it complex partners will find it very complex.

Reply:
For example we have `submittedForCustomerApproval`. Can we rename `submitted` and `approved` to mention they are scoped to admins.